plugins/mac-vth264/encoder.c

```diff
@@ -871,6 +916,18 @@ static obs_properties_t *vt_h264_properties(void *unused)
 	obs_properties_t *props = obs_properties_create();
 	obs_property_t *p;
 
+#ifdef __aarch64__
+	p = obs_properties_add_list(props, "rate_control", TEXT_RATE_CONTROL,
+				    OBS_COMBO_TYPE_LIST,
+				    OBS_COMBO_FORMAT_STRING);
+	obs_property_list_add_string(p, TEXT_RC_VBR, "VBR");
+	obs_property_list_add_string(p, TEXT_RC_QUALITY, "Constant quality");
+
+	obs_property_set_modified_callback(p, rate_control_modified);
+
+	p = obs_properties_add_int(props, "quality", TEXT_QUALITY, 1, 100, 1);
```

Comment:
From the limited testing I have done with a game I'm recording clips of sometimes, even a variation of 10 can make a difference (40 has limited artefacts while 50 is fine as an example), so I personally think the user should have granularity, but up to you anyway.

Reply:
> a variation of 10 can make a difference (40 has limited artefacts while 50 is fine as an example)

as much as I personally hate the text input, in that case it's probably fine to leave as is. But since you say taht 40 has some artifacts, do you think thats a good default value? I'm thinking in that case it should be bigger by default - again, I didn't test, I'm on Intel.
Speaking of which, I have yet to confirm myself but according to what I've heard this setting isn't new or exclusive to Apple Silicon - I'll test it myself tomorrow and get back to you.

Reply:
Maybe a compromise could be made by having a drop-down menu with 5 options (low/medium/high/lossless/custom), where if "custom" is selected, you can input the number in a textbox

Reply:
> But since you say taht 40 has some artifacts, do you think thats a good default value? I'm thinking in that case it should be bigger by default - again, I didn't test, I'm on Intel.
Maybe 50 would be a better default value, yeah. And that'd be surprising if it works on Intel as well, and probably worth informing the ffmpeg devs about. Let me know how it goes

Reply:
@gxalpha I take it OBS' current plugin API doesn't allow adding a slider to a property window? Because that (coupled with a decent default value) would be the best way to represent this range value IMO.

Reply:
It doesn't. If needed, I suppose it could be added, shouldn't be *too* difficult to do - though that depends on how familiar @AuroraWright is with the properties system (and properties-view), it might be out of scope for this PR and need to be retrofitted later.